packages/million/block.ts

```diff
@@ -46,7 +46,16 @@ export const block = (
   shouldUpdate?: (oldProps: MillionProps, newProps: MillionProps) => boolean,
   svg?: boolean,
 ) => {
-  const vnode = fn(HOLE_PROXY);
+  const vnode =
+    typeof fn === 'function' && !!fn.prototype
+      ? new (fn as any)(HOLE_PROXY).render()
+      : typeof fn === 'object'
+      ? new Proxy(fn, {
+          get(_, key: string): Hole {
+            return { $: key };
+          },
+        })
+      : fn(HOLE_PROXY);
```

Comment:
I don't understand how this allows for blocks to become cross file - could you add some context in your PR description about how this approach works?

This change doesn't make any changes to the compiler, so we'll still run into issues there.

Reply:
This is just to support class component for now